source/common/formatter/stream_info_formatter.cc

```diff
@@ -1428,6 +1428,26 @@ const StreamInfoFormatterProviderLookupTable& getKnownStreamInfoFormatterProvide
             [](const std::string& key, absl::optional<size_t> max_length) {
               return std::make_unique<EnvironmentFormatter>(key, max_length);
             }}},
+          {"UPSTREAM_CONNECTION_POOL_READY_DURATION",
+           {CommandSyntaxChecker::COMMAND_ONLY,
+            [](const std::string&, const absl::optional<size_t>&) {
+              return std::make_unique<StreamInfoDurationFormatterProvider>(
+                  [](const StreamInfo::StreamInfo& stream_info) {
+                    absl::optional<std::chrono::nanoseconds> result;
```

Comment:
yes, i also tried what you say first, but it can not be compiled correctly since the return type is not the same with absl::nullopt and absl::optional<std::chrono::nanoseconds>.

Reply:
I think you need to help the compiler out by putting `[](const StreamInfo::StreamInfo& stream_info) -> absl::optional<std::chrono::nanoseconds>` as type deduction might be failing

Reply:
get it, thanks, PTAL again. @KBaichoo

Reply:
And there are also the same code optimization in other formatter, maybe i will fix them all in a new PR.